### PR TITLE
Tickets/dm 13458 add antlr4 and mysql grammar to qserv build

### PR DESCRIPTION
**Pull request by n8pease**

### CODE REVIEW

File: core/modules/parser/SConscript [hunk 1]
```diff
@@ -17,15 +17,19 @@ antlrTgts = env.Antlr(['SqlSQL2.g'],
                       ANTLR_OUT='.',
                       )
 
+antlr4LexerTgt = env.Antlr4(['MySqlLexer.g4'], ANTLR4_OUT='.')
+                      
+antlr4ParserTgt = env.Antlr4(['MySqlLexer.tokens', 'MySqlParser.g4'], ANTLR4_OUT='.')
```
Comment (andy-slac) [inline core/modules/parser/SConscript]: Maybe add a comment here that `MySqlLexer.tokens` is on the target list from the previous line?

File: site_scons/site_tools/antlr4.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        env['ANTLR4_OUT'] = Dir(env['ANTLR4_OUT'])
+        flags.append('-o ${ANTLR4_OUT}')
+
+    # -lib flag
+    if env['ANTLR4_GLIB']:
+        env['ANTLR4_GLIB'] = Dir(env['ANTLR4_GLIB'])
+        flags.append('-lib ${ANTLR4_GLIB}')
+
+    # update antlr4 flags
+    env['ANTLR4_FLAGS'] = str(flags)
```
Comment (andy-slac) [inline site_scons/site_tools/antlr4.py]: Hmm, I think this is very bad idea, but it is copied from old Antlr tool so it is probably acceptable for now (until Fritz writes correct CMake version of it 😸).

File: site_scons/site_tools/antlr4.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    for src in source:
+        src = File(src)
+        stem = src.abspath
+        if stem.endswith(antlr4_token_suffix):
+            deps.append(src)
+            continue
+        grammars.append(src)
+        if stem.endswith(antlr4_suffix):
+            stem = stem[:-len(antlr4_suffix)]
+        kindInfo = lexerFiles if "Lexer" in stem else parserFiles
```
Comment (andy-slac) [inline site_scons/site_tools/antlr4.py]: Add comment here stating explicitly that it relies on a particular naming convention of g4 files?

File: site_scons/site_tools/antlr4.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        for kind, suffixes in kindInfo:
+            for ext in suffixes:
+                target.append(File(stem + kind + ext))
```
Comment (andy-slac) [inline site_scons/site_tools/antlr4.py]: Shorter version:

    target += [File(stem + kind + ext) for ext in suffixes]

File: site_scons/site_tools/antlr4.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        ANTLR4_HSUFFIX='.h',
+        ANTLR4_CCSUFFIX='.cpp',
+        ANTLR4_TOKENSUFFIX='.tokens',
+        # Antlr command
+        ANTLR4_COM="$ANTLR4 $ANTLR4_FLAGS $SOURCES",
+        ANTLR4_COMSTR='',
+        ANTLR4_GLIB='',
+    )
+    env['BUILDERS']['Antlr4'] = _antlr4_builder
+    env['ENV']['ANTLR4_DIR'] = os.environ['ANTLR4_DIR']
```
Comment (andy-slac) [inline site_scons/site_tools/antlr4.py]: Comment why is this needed (antlr4 script uses $ANTLR4_DIR)?

**Comment by fritzm** [comment]
Looks good to me!

BTW, the Travis builds here won't pass until we get a qserv:dev base container published that includes the antlr4 eups package, and that can't happen until this PR is merged (to bring in the add to the qserv eups table file), so catch-22...  I'm okay with this merging to master as soon as it gets a clean Jenkins CI run on all platforms.
